backend/tests/youi-adapter.spec.ts

```diff
@@ -0,0 +1,90 @@
+import { expect } from "chai";
+import { YouiEvents } from '../src/youi-events';
+import { IMethod, IPromiseCallbacks, IRpc } from "@sap-devx/webview-rpc/out.ext/rpc-common";
+import { GeneratorFilter } from "../src/filter";
+import { AppWizard } from "@sap-devx/yeoman-ui-types";
+import { YouiAdapter } from '../src/youi-adapter';
+import { YeomanUI } from "../src/yeomanui";
+import messages from "../src/messages";
+
+describe('YouiAdapter', () => {
+
+    class TestEvents implements YouiEvents {
+	public doGeneratorDone(success: boolean, message: string, selectedWorkspace: string, type: string, targetPath?: string): void {
+		return;
+	}
+	public doGeneratorInstall(): void {
+		return;
+	}
+	public showProgress(message?: string): void {
+		return;
+	}
+	public getAppWizard(): AppWizard {
+		return;
+	}
+	public executeCommand(command: string, ...rest: any[]): Thenable<any> {
+		return;
+	}
+    }
+    
+    class TestRpc implements IRpc {
+	public timeout: number;
+	public promiseCallbacks: Map<number, IPromiseCallbacks>;
+	public methods: Map<string, IMethod>;
+	public sendRequest(): void {
+		return;
+	}
+	public sendResponse(): void {
+		return;
+	}
+	public setResponseTimeout(): void {
+		return;
+	}
+	public registerMethod(): void {
+		return;
+	}
+	public unregisterMethod(): void {
+		return;
+	}
+	public listLocalMethods(): string[] {
+		return [];
+	}
+	public handleResponse(): void {
+		return;
+	}
+	public listRemoteMethods(): Promise<string[]> {
+		return Promise.resolve([]);
+	}
+	public invoke(): Promise<any> {
+		return Promise.resolve();
+	}
+	public handleRequest(): Promise<void> {
+		return Promise.resolve();
+	}
+    }
+
+    const testLogger = {
+	debug: () => true, error: () => true,
+	fatal: () => true, warn: () => true, info: () => true, trace: () => true, getChildLogger: () => { return testLogger; }
+    };
+
+    const rpc = new TestRpc();
+    const outputChannel: any = {
+	appendLine: () => "",
+	show: () => ""
+    };
+    
+    const youiEvents = new TestEvents();
+    
+    const yeomanUi: YeomanUI = new YeomanUI(rpc, youiEvents, outputChannel, testLogger,
+	{ filter: GeneratorFilter.create(), messages }, undefined);
+
+    describe('#prompt()', () => {
+        it('passes null call back', async () => {
+            const youiAdapter = new YouiAdapter(youiEvents, outputChannel); 
+            youiAdapter.setYeomanUI(yeomanUi);
+            const response = await youiAdapter.prompt([], null);
+            expect(response).to.be.empty;
+        });
+    });
+});
```

Comment:
it's doing the job. but can we test not empty question obj ?
const questions = [{ name: "q1" }];

rpc.invoke can return --> 
{
					firstName: "john",
					lastName: "doe"
				};